maxfw/tests/test_image_utils.py

```diff
@@ -0,0 +1,48 @@
+# Standard libs
+import io
+import os
+
+# Dependencies
+import pytest
+from PIL import Image
+
+# The module to test
+from maxfw.utils.image_utils import ImagePreprocessor, ImagePostprocessor
+
+# Initialize a test input file
+stream = io.BytesIO()
+Image.open('maxfw/tests/test_image.jpg').convert('RGBA').save(stream, 'PNG')
+test_input = stream.getvalue()
+
+def test_imagepreprocessor():
+    '''
+    Test the ImagePreprocessor.
+    '''
+
+    # Test reshape
+    preprocessor = ImagePreprocessor(rotate_angle=180,resize_shape=(100,100), verbose=True, normalize=False)
+    assert preprocessor.preprocess_imagedata(test_input).shape == (100, 100, 3)
+
+    # Test grayscale
+    preprocessor = ImagePreprocessor(resize_shape=(100, 100), grayscale=True, verbose=True)
+    assert preprocessor.preprocess_imagedata(test_input).shape == (100, 100)
+
+    # test file output
+    preprocessor.preprocess_imagedata(test_input, png_file_path='test_output.png')
```

Comment:
I think these tests should probably be split up into different test functions so it's easier to see which specific test cases fail. For example if 3 out of 5 tests are passing for the preprocessor. Currently, we'd only see 1 test fail if a one or more functions are broken.

Reply:
I think we need to add more tests so that each test tests a single operation.

Reply:
+1 on that!

Reply:
> +1 on that!

will try to add those myself

Reply:
Added individual test cases in 29426fa.